staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -1719,16 +1722,34 @@ func TestValidationExpressions(t *testing.T) {
 					if celValidator == nil {
 						t.Fatal("expected non nil validator")
 					}
-					errs, _ = celValidator.Validate(field.NewPath("root"), &s, tt.obj, tt.costBudget)
+					errs, _ = celValidator.Validate(field.NewPath("root"), &s, tt.obj, tt.costBudget, ctx)
 					for _, err := range errs {
-						if err.Type == field.ErrorTypeInvalid && strings.Contains(err.Error(), "call cost exceeds limit for rule") {
+						if err.Type == field.ErrorTypeInvalid && strings.Contains(err.Error(), "operation cancelled: actual cost limit exceeded") {
```

Comment:
I'm surprised this changed for a test case that isn't exercising a cancelled context

Reply:
I have raised a pr to update the error handling: https://github.com/kubernetes/kubernetes/pull/108695
Previous we handle specific error got from cel such like `no such overload` separately because it could be confusing for Kubernetes users. Now with cel updates their error message, there is no need for us to wrap it anymore. `operation cancelled: actual cost limit exceeded` should be clear to the Kubernetes user with specific rule also returned together.

Reply:
[commented](https://github.com/kubernetes/kubernetes/pull/108695/files#r826506419) on that PR, but I'm wondering if we're correctly handling the case where a single call exceeded the per-call limit, but didn't exhaust the overall budget

Reply:
coming back to this change... would this test fail in this PR without this change? I'm trying to understand if the message changed as a result of this PR, or if this was just cleanup

Reply:
I have reverted the change here and will handle it in #108695. To my understanding, if the cost limit exceed for one expression but the overall cost limit does not exceed, we will keep validating the following rules and return error for this specific failed rule to indicate this rule exceed per expression limit. The CR validation as a whole sure will fail but all the following rules get the chance to be validated and we will return all possible errors together back to user.